pyplanet/apps/contrib/live_rankings/__init__.py

```diff
@@ -109,27 +109,57 @@ async def handle_scores(self, players):
+						new_ranking = dict(login=player['login'], nickname=player['name'], score=player['mappoints'], score_matchpoints=player['matchpoints'], points_added=0)
 						self.current_rankings.append(new_ranking)
-
+		elif 'cup' in current_script or 'trackmania/tm_cup_online' in current_script:
+			if section == 'EndRound':
+				mode_settings = await self.instance.mode_manager.get_settings()	
+				pointlimit = mode_settings['S_PointsLimit']
+				for player in players:
+					if 'map_points' in player:
+						if player['map_points'] != -1 and player['match_points'] == pointlimit:
+							new_ranking = dict(login=player['player'].login, nickname=player['player'].nickname, score=player['map_points'], score_matchpoints='Finalist', points_added=0)
+							self.current_rankings.append(new_ranking)
+						elif player['map_points'] != -1 and player['match_points'] > pointlimit:
+							new_ranking = dict(login=player['player'].login, nickname=player['player'].nickname, score=player['map_points'], score_matchpoints='Winner', points_added=0)
+							self.current_rankings.append(new_ranking)
+						else:
+							new_ranking = dict(login=player['player'].login, nickname=player['player'].nickname, score=player['map_points'], score_matchpoints=player['match_points'], points_added=0)
+							self.current_rankings.append(new_ranking)
+					elif 'mappoints' in player:
+						if player['mappoints'] != -1 and player['matchpoints'] == pointlimit:
+							new_ranking = dict(login=player['login'], nickname=player['name'], score=player['mappoints'], score_matchpoints='$wFinalist', points_added=0)
+							#self.current_rankings.append(new_ranking)
+						elif player['mappoints'] != -1 and player['matchpoints'] > pointlimit:
+							new_ranking = dict(login=player['login'], nickname=player['name'], score=player['mappoints'], score_matchpoints='$wWinner', points_added=0)
+							#self.current_rankings.append(new_ranking)
+						else:
+							new_ranking = dict(login=player['login'], nickname=player['name'], score=player['mappoints'], score_matchpoints=player['matchpoints'], points_added=0)
+							self.current_rankings.append(new_ranking)
+		
 			self.current_rankings.sort(key=lambda x: x['score'])
 			self.current_rankings.reverse()
 
 	async def map_start(self, map, restarted, **kwargs):
-		self.current_rankings = []
-		self.current_finishes = []
-		self.dedimania_enabled = ('dedimania' in self.instance.apps.apps and 'dedimania' not in self.instance.apps.unloaded_apps)
-		await self.get_points_repartition()
-		await self.widget.display()
+			self.dedimania_enabled = ('dedimania' in self.instance.apps.apps and 'dedimania' not in self.instance.apps.unloaded_apps)
+			
+			if restarted == False:
+				await self.get_points_repartition()
+				await self.widget.display()
+			else:
+				self.current_rankings = []
+				self.current_finishes = []
+				await self.get_points_repartition()
+				await self.widget.display()
```

Comment:
I guess we could move the reset of the rankings and finishes to a 'restarted == True' if and get the repartition and display the widget outside the if.

Why shouldn't the rankings and finishes not be reset when moving to the next map though?

Reply:
Well it depends on the following things:

I have seen streamers that use stream Cup mode with pyplanet and they need to press TAB each time to show the total points gathered during x maps of a cup. It really looks obnoxious to viewers to see the ScoreTable and the match is still proceeding.

For Players its also quite useful to know what points they gave gathered not only during 1 map but also on the whole match. 

In Cup mode you play certain maps till pointlimit is reached and when player becomes finalist he/she/it needs 1 win to be Winner.

Problem now actually is you as player/streamer only the scores on one map and then you restart fresh on the next map even though there is no winner which can be confusing for players/streamers/viewers.

In this way with this addition players can always see with a quick glance without using tab. 

(More Competitions with more information about the match ongoing keeps people interested a.k.a. more exposure to the game for instance) Yes i know pyplanet needs to be clean. But its a way to keep people interested in the game/games.

And regarding to what is what in the widget see this example:

```
<frame pos="0 -10">
	<quad pos="-56 4" z-index="0" size="4 3" bgcolor="00000070"/>
	<label pos="-56 2" z-index="1" size="4 3" text="Pos" textsize="0.2" textfont="RajdhaniMono"  textemboss="1" halign="left" valign="center2"/>
	<quad pos="-51 4" z-index="0" size="20.5 3" bgcolor="00000070"/>
	<label pos="-42 2" z-index="1" size="20 3" text="Nick" textsize="0.2" textfont="RajdhaniMono"  textemboss="1" halign="left" valign="center2"/>
	<quad pos="-30 4" z-index="0" size="10 3" bgcolor="00000070"/>
	<label pos="-28 2" z-index="1" size="20 3" text="Round" textsize="0.2" textfont="RajdhaniMono"  textemboss="1" halign="left" valign="center2"/>
	<quad pos="-19 4" z-index="0" size="10 3" bgcolor="00000070"/>
	<label pos="-17 2" z-index="1" size="20 3" text="Match" textsize="0.2" textfont="RajdhaniMono"  textemboss="1" halign="left" valign="center2"/>
	
	<quad pos="-56 0" z-index="0" size="4 3" bgcolor="00000070"/>
	<quad pos="-51 0" z-index="0" size="20.5 3" bgcolor="00000070"/>
	<label pos="-42 -2" z-index="1" size="20 3" text="$fffw1lla" textsize="0.2" textfont="RajdhaniMono"  textemboss="1" halign="left" valign="center2"/>
	<quad pos="-30 0" z-index="0" size="10 3" bgcolor="00000070"/>
	<label pos="-25 -2" size="7 3" z-index="1" textsize="0.2" halign="center" valign="center2" text="$fff 10" textfont="RajdhaniMono"  textemboss="1"/>
	<quad pos="-19 0" z-index="0" size="10 3" bgcolor="00000070"/>
	<label pos="-15 -2" size="7 3" z-index="1" textsize="0.2" halign="center" valign="center2" text="$fff 100" textfont="RajdhaniMono"  textemboss="1"/>
	
</frame>
```

<img width="480" alt="2021-02-21 (3)" src="https://user-images.githubusercontent.com/1784761/108618668-344a6b80-7420-11eb-8e48-3a281a669d62.png">

Yes screenshot is made in the title tools just as a simple way to show information really quick.

For Rounds/Team mode it might be different regarding the setup of tournament or cup i agree.

But it could be handy to see how many points players gathered during matches/maps. Yes if you play 1 map is useful but not if you have multiple maps and the round scores get reset but there is no way to see the total matchpoints gathered. 
For viewers/streamers the interest might lose, and people will disregard the game as quite less information is provided.

(Yes Nadeo should do that in fact......)

Regarding the Restarted == false:

Maps on cup mode if Restarted == true it makes a new Match so a better check would be to be for gamemodes for instance.

current_rankings is also for matchpoints so finishes could be restarted.

Sometimes simplicity is the key and yes people want to be information fed.